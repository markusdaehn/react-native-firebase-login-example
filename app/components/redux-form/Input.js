import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { FormInput } from 'react-native-elements';

export default class Input extends Component {
  constructor(props) {
    super(props);
  }

  static propTypes = {
    input: PropTypes.shape({
      onChange: PropTypes.func.isRequired,
      onBlur: PropTypes.func.isRequired,
      onFocus: PropTypes.func.isRequired,
    }).isRequired,
  }

  focus() {
    return this.input.focus();
  }

  render() {
    let { input, meta, ...otherProps } = this.props;
    return (
        <FormInput
          ref={(input) => this.input = input}
          {...input}
          {...otherProps}
        />
    );
  }
}
