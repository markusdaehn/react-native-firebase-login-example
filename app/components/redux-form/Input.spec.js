import 'react-native';
import React from 'react';
import Input from './Input';
import { shallow, mount } from 'enzyme';

describe('Input', function () {
  let ev = {target: {value: 'new value'}};
  let input, value, inputWrapper;

  beforeEach(function () {
    value = 'The value';
    input = {
      onChange: jest.fn(),
      onBlur: jest.fn(),
      onFocus: jest.fn(),
      value,
    };
  });
  describe('when rendered', function () {
    beforeEach(function () {
      inputWrapper = shallow(<Input input={input} a='a' b='b'/>);
    });
    it('should render correctly', function () {
      expect(inputWrapper).toMatchSnapshot();
    });
    describe('and when the text is changed', function () {
      beforeEach(function () {
        inputWrapper.find('FormInput').simulate('change', ev);
      });
      it('should call the input.onChange() method with the expected event data', function () {
        expect(input.onChange).toBeCalledWith(ev);
      });
    });
    describe('and when input is navigated away', function () {
      beforeEach(function () {
        inputWrapper.find('FormInput').simulate('blur', {target: {value: 'new value'}});
      });
      it('should call the input.onBlur() method', function () {
        expect(input.onBlur).toBeCalled();
      });
    });
    describe('and when input gains focus', function () {
      beforeEach(function () {
        inputWrapper.find('FormInput').simulate('focus', {target: {value: 'new value'}});
      });
      it('should call the input.onFocus() method', function () {
        expect(input.onFocus).toBeCalled();
      });
    });
  });
});
