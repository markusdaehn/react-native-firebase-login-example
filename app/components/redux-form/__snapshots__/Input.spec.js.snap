// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input when rendered should render correctly 1`] = `
<FormInput
  a="a"
  b="b"
  normalizeFontSize={true}
  onBlur={[Function]}
  onChange={[Function]}
  onChangeText={[Function]}
  onFocus={[Function]}
  value="The value"
/>
`;
